cdap-docs/developers-manual/source/pipelines/developing-plugins/presentation-plugins.rst

```diff
@@ -0,0 +1,771 @@
+          "widget-attributes": {
+            "default": "1",
+            "min": "1",
+            "max": "100"
+          }
+        }
+     
+   * - ``password``
+     - No attributes
+     - ``string``
+     - Default HTML password entry box
+     - ::
+
+        {
+          "name": "property-password",
+          "label": "My Password Property",
+          "widget-type": "password",
+          "widget-attributes": {}
+        }
+     
+   * - ``schema``
+     - - ``schema-default-type``: default type for each newly-added field in the schema
+       - ``schema-types``: list of schema types for each field from which the user can chose when setting the schema
+     - ``string``
+     - A four-column, editable table for representing the schema of a plugin
+     - ::
+
+        {
+          "name": "property-schema",
+          "label": "My Schema Property",
+          "widget-type": "schema",
+          "widget-attributes": {
+            "schema-default-type": "string",
+            "schema-types": [
+              "boolean",
+              "int",
+              "long",
+              "float",
+              "double",
+              "bytes",
+              "string",
+              "map<string, string>"
+            ]
+          }
+        }
+     
+   * - ``select``
+     - - ``default``: default value from the list
+       - ``values``: list of values for the drop-down
+     - ``string``
+     - An HTML drop-down with a list of values; allows one choice from the list
+     - ::
+
+        {
+          "name": "property-select",
+          "label": "My Select Property",
+          "widget-type": "select",
+          "widget-attributes": {
+              "default": "Bananas",
+              "values": ["Apples", "Oranges", "Bananas"]
+          }
+        }
+     
+   * - ``textarea``
+     - - ``default``: default value for the widget
+       - ``rows``: height of the ``textarea``
+     - ``string``
+     - An HTML ``textarea`` element which accepts a default value attribute and a height in rows
+     - ::
+
+        {
+          "name": "property-textarea",
+          "label": "My Textarea Property",
+          "widget-type": "textarea",
+          "widget-attributes": {
+            "default": "Default text.",
+            "rows": "1"
+          }
+        }
+     
+   * - ``textbox``
+     - ``default``: default value for the widget
+     - ``string``
+     - An HTML textbox, used to enter any string, with a default value attribute
+     - ::
+
+        {
+          "name": "property-textbox",
+          "label": "My Textbox Property",
+          "widget-type": "textbox",
+          "widget-attributes": {
+            "default": "Default text."
+          }
+        }
+     
+
+.. _plugins-presentation-plugin-function:
+
+Plugin Function
+---------------
+A plugin function is a method exposed by a particular plugin that can be used to fetch an
+output schema for a plugin. These fields need to be configured to use the plugin functions
```

Comment:
it can be used for other things too. currently, the ones we have only fetch output schema